cache/watcher.go

```diff
@@ -15,30 +15,33 @@
 package cache
 
 import (
+	"sync"
 	"sync/atomic"
 
+	"go.etcd.io/etcd/api/v3/v3rpc/rpctypes"
 	clientv3 "go.etcd.io/etcd/client/v3"
 )
 
 // watcher holds one client’s buffered stream of events.
 type watcher struct {
-	eventQueue chan []*clientv3.Event
-	keyPred    KeyPredicate
-	stopped    int32
-	done       chan struct{} // closed together with Stop()
+	respQueue chan clientv3.WatchResponse
+	keyPred   KeyPredicate
+	stopped   int32
+	done      chan struct{} // closed together with Stop()
+	mu        sync.Mutex
```

Comment:
What does mux protect?

Please move mutex to separate section of variables and put variables protected by mutext under mutex.